pkg/csi/cinder/controllerserver.go

```diff
@@ -1033,7 +1038,24 @@ func (cs *controllerServer) ControllerExpandVolume(ctx context.Context, req *csi
 	}, nil
 }
 
-func getCreateVolumeResponse(vol *volumes.Volume, volCtx map[string]string, ignoreVolumeAZ bool, accessibleTopologyReq *csi.TopologyRequirement) *csi.CreateVolumeResponse {
+func getTopology(vol *volumes.Volume, topologyReq *csi.TopologyRequirement, ignoreVolumeAZ bool) []*csi.Topology {
+	var accessibleTopology []*csi.Topology
+	if ignoreVolumeAZ {
+		if topologyReq != nil {
+			accessibleTopology = topologyReq.GetPreferred()
+		}
+	} else {
```

Comment:
I think you've lost your previous change which refused to create a segment with an empty AZ.